src/RequestHandler.cpp

```diff
@@ -173,10 +192,28 @@ void RequestHandler::consumePartitions(KafkaMessageMetadataStruct &MessageData,
     EOFPartitionCounter++;
 }
 
+/// Prints and formats message Metadata.
+///
+/// \param MessageData
 void RequestHandler::printMessageMetadata(
     KafkaMessageMetadataStruct &MessageData) {
   std::cout << fmt::format("\n{:_>67}{:>67}\nTimestamp: {:>11} || PartitionID: "
                            "{:>5} || Offset: {:>7}\n",
                            "\n", "|", MessageData.Timestamp,
                            MessageData.Partition, MessageData.Offset);
 }
+
+/// Calculates topic's lowest offset and subscribes to it to print the entire
+/// topic.
+///
+/// \param TopicName
+void RequestHandler::printEntireTopic(const std::string &TopicName) {
+  std::vector<OffsetsStruct> OffsetsStruct =
+      KafkaConnection->getTopicsHighAndLowOffsets(TopicName);
+  int64_t MinOffset = OffsetsStruct[0].LowOffset;
+  for (auto Struct : OffsetsStruct) {
```

Comment:
Should change this just in case of confusion with the c++ type, even if just to `OffsetStruct`